expression/scalar_function.go

```diff
@@ -250,6 +253,25 @@ func (sf *ScalarFunction) EvalJSON(ctx context.Context, row types.Row) (json.Bin
 	return sf.Function.evalJSON(row)
 }
 
+// HashCode implements Expression interface.
+func (sf *ScalarFunction) HashCode(sc *stmtctx.StatementContext) []byte {
+	if len(sf.hashcode) > 0 {
+		return sf.hashcode
+	}
+	v := make([]types.Datum, 0, len(sf.GetArgs())+1)
+	var err error
+	sf.hashcode, err = codec.EncodeValue(sc, nil, types.NewStringDatum(sf.FuncName.L))
```

Comment:
Any good suggestion? This is just some code i recovered since it has been deleted once.
I came up with the idea that:
```
Constant: ConstantFlag+encoded value
Column: ColumnFlag+encoded value
ScalarFunction: SFFlag+encoded function name + 0(or other flag byte) + encoded arg_1 + 0 + ... 
```
This way column and scalarFunction can call `EncodeBytes([]byte)` directly.

Reply:
for ScalaFunction: "SFFlag+encoded function name + encoded arg0 + encoded arg1 ... " is distinguishable ?

Reply:
If we use `EncodeBytes`, the returned value only contains encoded value, no flag byte exists.
So if we don't add flag byte between the encoded args, there may be the case that different args have the same encoded value?